packages/components/src/internal/components/domainproperties/TextChoiceOptions.tsx

```diff
@@ -65,42 +80,89 @@ export const TextChoiceOptionsImpl: FC<ImplProps> = memo(props => {
         validValues,
         replaceValues,
         maxValueCount = MAX_VALID_TEXT_CHOICES,
+        lockedForDomain,
     } = props;
     const [selectedIndex, setSelectedIndex] = useState<number>();
     const [currentValue, setCurrentValue] = useState<string>();
+    const [currentError, setCurrentError] = useState<string>();
     const [showAddValuesModal, setShowAddValuesModal] = useState<boolean>();
-    const currentInUse = fieldValues.indexOf(currentValue) > -1;
+
+    // keep a map from the updated values for the in use field values to their original values
+    const [fieldValueUpdates, setFieldValueUpdates] = useState<Record<string, string>>({});
+    useEffect(() => {
+        setFieldValueUpdates(
+            Object.keys(fieldValues).reduce((prev, current) => {
+                prev[current] = current;
+                return prev;
+            }, {})
+        );
+    }, [fieldValues]);
+
+    const selectedValue = useMemo(() => validValues[selectedIndex], [validValues, selectedIndex]);
+    const currentInUse = fieldValueUpdates.hasOwnProperty(selectedValue);
+    const currentLocked =
+        currentInUse && (lockedForDomain || (fieldValues[fieldValueUpdates[selectedValue]]?.locked ?? false));
+
+    const isValueDuplicate = useCallback(
+        (val: string): boolean => {
+            return validValues.indexOf(val) !== -1;
+        },
+        [validValues]
```

Comment:
This looks like it won't catch duplicates of values that are being updated/added.  Is that handled already as part of saving?

Reply:
This `isValueDuplicate` is used for those cases where the single input value is being applied and for the add values modal case. So I believe it covers those.